api/osb/logic.go

```diff
@@ -0,0 +1,181 @@
+}
+
+func (b *BusinessLogic) Deprovision(request *osbc.DeprovisionRequest, c *broker.RequestContext) (*broker.DeprovisionResponse, error) {
+	client, err := b.osbClient(c.Request)
+	if err != nil {
+		return nil, err
+	}
+	response, err := client.DeprovisionInstance(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &broker.DeprovisionResponse{
+		DeprovisionResponse: *response,
+	}, nil
+}
+
+func (b *BusinessLogic) LastOperation(request *osbc.LastOperationRequest, c *broker.RequestContext) (*broker.LastOperationResponse, error) {
+	client, err := b.osbClient(c.Request)
+	if err != nil {
+		return nil, err
+	}
+	response, err := client.PollLastOperation(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &broker.LastOperationResponse{
+		LastOperationResponse: *response,
+	}, nil
+}
+
+func (b *BusinessLogic) Bind(request *osbc.BindRequest, c *broker.RequestContext) (*broker.BindResponse, error) {
+	client, err := b.osbClient(c.Request)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := client.Bind(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &broker.BindResponse{
+		BindResponse: *response,
+	}, nil
+
+}
+
+func (b *BusinessLogic) Unbind(request *osbc.UnbindRequest, c *broker.RequestContext) (*broker.UnbindResponse, error) {
+	client, err := b.osbClient(c.Request)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := client.Unbind(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &broker.UnbindResponse{
+		UnbindResponse: *response,
+	}, nil
+}
+
+func (b *BusinessLogic) Update(request *osbc.UpdateInstanceRequest, c *broker.RequestContext) (*broker.UpdateInstanceResponse, error) {
+	client, err := b.osbClient(c.Request)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := client.UpdateInstance(request)
+	if err != nil {
+		return nil, err
+	}
+
+	return &broker.UpdateInstanceResponse{
+		UpdateInstanceResponse: *response,
+	}, nil
+}
+
+func (b *BusinessLogic) ValidateBrokerAPIVersion(version string) error {
+	expectedVersion := osbc.LatestAPIVersion().HeaderValue()
+	if version != expectedVersion {
+		return fmt.Errorf("error validating OSB Version: expected %s but was %s", expectedVersion, version)
+	}
+	return nil
+}
+
+func clientConfigForBroker(broker *types.Broker) *osbc.ClientConfiguration {
+	config := osbc.DefaultClientConfiguration()
+	config.Name = broker.Name
+	config.URL = broker.URL
+	config.AuthConfig = &osbc.AuthConfig{
+		BasicAuthConfig: &osbc.BasicAuthConfig{
+			Username: broker.User,
+			Password: broker.Password,
+		},
+	}
+	return config
+}
+
+func (b *BusinessLogic) osbClient(request *http.Request) (osbc.Client, error) {
+	vars := mux.Vars(request)
+	brokerID, ok := vars[BrokerIDPathParam]
+	logrus.Debugf("Obtained path parameter [brokerID = %s] from mux vars", brokerID)
+	if !ok {
+		return nil, fmt.Errorf("Error creating OSB client: brokerID path parameter not found")
+	}
+	broker, err := b.brokerStorage.Find(request.Context(), brokerID)
```

Comment:
Is the context object really used?

Reply:
no it's not mandatory - the idea was to make the interfaces pass it from now in case we need them in the future (so we don't have to change all the interfaces later).